### PR TITLE
Add mariadb

### PR DESCRIPTION
**Pull request by tehut**

### CODE REVIEW

File: incubator/mariadb/maria.libsonnet [hunk 1]
```diff
@@ -0,0 +1,321 @@
+local k = import 'ksonnet.beta.2/k.libsonnet';
+local deployment = k.extensions.v1beta1.deployment;
+
+{
+  parts:: {
+
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: Extra newline

File: incubator/mariadb/maria.libsonnet [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    svc(namespace, name, metricsEnabled=true, labels={app:name}, selector={app:name})::
+      local defaults = {
+        serviceType: "ClusterIP"
+        // i could totally hardcode this and skip the defaults object
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: Either is fine.

File: incubator/mariadb/maria.libsonnet [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+      metadata: {
+        name: name,
+        namespace: namespace,
+        labels: labels,
+      },
+      type: "Opaque",
+      data: {
+         "mariadb-root-password":
+         if mariaRootPassword != null then std.base64(mariaRootPassword)
+          else error "Please add a root password",
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: These should never be null. You can remove these conditionals.

File: incubator/mariadb/maria.libsonnet [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+          ],
+          resources: {
+            requests: {
+              storage: defaults.size,
+            },
+          },
+          [if storageClassName != null then "storageClass"]:storageClassName,
+        },
+      },
+    deployment:: {
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: Missing newline between `pvc` and `deployment`

File: incubator/mariadb/maria.libsonnet [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+            "prometheus.io/port": "9104",
+          },
+        },
+        mariaConfig: {
+          user: "",
+          db: "",
+        },
+      },
+
+      persistent(namespace, name, passwordSecretName, mariaConfig=defaults.mariaConfig, metricsEnabled= true, existingClaim=name, labels={app:name})::
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: `metricsEnabled= true` weird spacing

File: incubator/mariadb/maria.libsonnet [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+        {
+          name: "MARIADB_PASSWORD",
+          valueFrom: {
+            secretKeyRef: {
+              name: passwordSecretName,
+              key: "mariadb-password",
+            },
+          },
+        }
+      ],
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet:175]: This block has weird indentation.

File: incubator/mariadb/maria.libsonnet [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+          valueFrom: {
+            secretKeyRef: {
+              name: passwordSecretName,
+              key: "mariadb-root-password",
+            },
+          },
+        },
+          // an error in the deployment.jsonnet file kept the MariadbUser container from being
+          // included in the original yaml.
+          // this container is expected tin the diff
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: Ok, you can remove this comment.

File: incubator/mariadb/maria.libsonnet [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+          if !metricsEnabled then []
+          else [
+            {
+              name: "metrics",
+              image: "%s:%s" % [defaults.metrics.image, defaults.metrics.imageTag],
+              imagePullPolicy: defaults.metrics.imagePullPolicy,
+              env: [
+                    // If we want to have an insecure version sketched out as an overlay we
+                    // would remove this object and use the insecure env
+                    // formerly :  if values.usePassword then ([
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: Indentation is weird here.

File: incubator/mariadb/maria.libsonnet [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+                        mountPath: "/bitnami/mariadb",
+                      },
+                    ],
+                  },
+                ] + metricsContainer,
+                volumes: [
+                  {
+                    name: "config",
+                    configMap: {
+                      name: name,
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: This needs to be a parameter that defaults to `name`, I think.

File: incubator/mariadb/maria.libsonnet [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+                    },
+                  }
+                ],
+              },
+            },
+          },
+        },
+    },
+  }
+}
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: Add newline at the end of file.

File: incubator/mariadb/maria.libsonnet [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+          valueFrom: {
+            secretKeyRef: {
+              name: passwordSecretName,
+              key: "mariadb-password",
+            },
+          },
+        }
+      ],
+
+      local base(namespace, name, passwordSecretName, mariaConfig=defaults.mariaConfig, metricsEnabled, claimName, labels) =
```
Comment (hausdorff) [inline incubator/mariadb/maria.libsonnet]: `mariaConfig` should not have a default value.

**Review by hausdorff — CHANGES_REQUESTED** [review]

File: incubator/mariadb/maria.libsonnet (hunk 1 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 2 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 3 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 4 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 5 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 7 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 9 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 8 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

File: incubator/mariadb/maria.libsonnet (hunk 10 above)
Comment (tehut) [reply incubator/mariadb/maria.libsonnet]: depressing checkmark

**Review by hausdorff — APPROVED** [review]
LGTM

**Pull request merged**